src/storages/file_storage.rs

```diff
@@ -88,14 +90,20 @@ impl Storage for FileStorage {
         // bytes to the end of a file.
         let mut file = OpenOptions::new()
             .write(true)
-            .append(true)
+            .append(false)
```

Comment:
What will happen if you turn off preallocation? I guess it will fail.

Reply:
It shouldn't fail (at least as far I covered it in tests; maybe there're some corner cases that I don't know about): in the next few lines I'm seeking to the offset